src/index.ts

```diff
@@ -67,6 +76,10 @@ export const beautifier: Beautifier = {
         rules: finalOptions,
       };
       const cli: CLIEngine = new CLIEngine(cliOptions);
+      // tslint:disable-next-line:promise-must-complete
```

Comment:
This may be a bug with https://github.com/Microsoft/tslint-microsoft-contrib/
If you can create a much smaller test case, you could report an issue 👍 .

Reply:
It looks like any `if` is a branch, and doesn't bother looking what it actually does to consider whether the promise completes or not (since we are using `return new Promise<string>((resolve, reject) => {`).

Maybe it's safer to remove that block and change the returns at the bottom to `return Promise.resolve`.

Reply:
Let's not make any major changes yet. We can try different approaches in a later PR 👍 .

Reply:
@Glavin001 would you mind creating a small test case and make an issue there? I think you can way better explain that error case.